### PR TITLE
Add compatibility to API 10.04

### PR DESCRIPTION
**Pull request by JM2K69**
* Add mandatory parameter to Connect-ArubaCX in second position $API
* This function return an $Script:Api for use in other functions 
* Add 3 functions , for Users, Firmware, Vlan

### CODE REVIEW

File: PowerArubaCX/Public/Firmware.ps1 [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      .SYNOPSIS
+      Get Aruba CX firmware
+
+      .DESCRIPTION
+      Get all informations about Aruba CX firmware
+
+      .EXAMPLE
+      Get-ArubaCXfirmware
+
+        current_version   : Virtual.10.04.0001
```
Comment (alagoutte) [inline PowerArubaCX/Public/Firmware.ps1]: No need to make a example of display but a description of function (with parameter when it is possible)

File: PowerArubaCX/Public/Connection.ps1 [hunk 2]
```diff
@@ -103,7 +106,9 @@ function Connect-ArubaCX {
         $connection
     }
 
-    End {
+    End 
+    {
+        $Script:Api = $Api
```
Comment (alagoutte) [inline PowerArubaCX/Public/Connection.ps1]: move to global variable

File: PowerArubaCX/Public/User.ps1 [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    <#
+      .SYNOPSIS
+      Get Aruba CX user
+
+      .DESCRIPTION
+      Get all informations about Aruba CX Users
+
+      .EXAMPLE
+      Get-ArubaCXusers
```
Comment (alagoutte) [inline PowerArubaCX/Public/User.ps1:11]: samewhere

File: PowerArubaCX/Public/interfaces.ps1 [hunk 4]
```diff
@@ -47,14 +47,15 @@ function Get-ArubaCXinterfaces {
...
-        $uri = "rest/v1/system/interfaces"
+        $uri = "rest/$Script:Api/system/interfaces"
 
         if ( $PsBoundParameters.ContainsKey('interface') ) {
           $interface = $interface -replace '/', '%2F'
           $uri += "/$interface"
         }
 
         $response = Invoke-ArubaCXRestMethod -uri $uri -method 'GET' @invokeParams
+        $response = $response -replace '%2F','/'
```
Comment (alagoutte) [inline PowerArubaCX/Public/interfaces.ps1]: don't replace, it is will complicated when pipeline...

File: PowerArubaCX/Public/ports.ps1 [hunk 5]
```diff
@@ -51,7 +51,7 @@ function Get-ArubaCXPorts {
             $invokeParams.add( 'attributes', $attributes )
         }
 
-        $uri = "rest/v1/system/ports"
+        $uri = "rest/$Script:Api/system/ports"
```
Comment (alagoutte) [inline PowerArubaCX/Public/ports.ps1]: i think move $API directly on Invoke-ArubaCXRestMethod

File: PowerArubaCX/Public/Connection.ps1 [hunk 6]
```diff
@@ -85,7 +88,7 @@ function Connect-ArubaCX {
         }
 
         $postParams = @{username = $Credentials.username; password = $Credentials.GetNetworkCredential().Password}
-        $url = "https://${Server}/rest/v1/login"
```
Comment (alagoutte) [inline PowerArubaCX/Public/Connection.ps1:88]: kept v1 for login

File: PowerArubaCX/Public/Connection.ps1 [hunk 7]
```diff
@@ -103,7 +106,9 @@ function Connect-ArubaCX {
         $connection
     }
```
Comment (alagoutte) [inline PowerArubaCX/Public/Connection.ps1:108]: Add get supported API release (there is a call for this)

File: PowerArubaCX/Public/User.ps1 [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+        admin
+        -----
+        /rest/v10.04/system/users/admin
+    #>
+    Param(
```
Comment (alagoutte) [inline PowerArubaCX/Public/User.ps1:16]: no parameter ? depth or other ?

File: PowerArubaCX/Public/vlan.ps1 [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+      .SYNOPSIS
+      Get list of all Aruba CX Vlan
+
+      .DESCRIPTION
+      Get list of all Aruba CX Vlan (port, lag, vlan... with name, IP Address, description)
+
+      .EXAMPLE
+      Get-ArubaCXvlan
+
+        1                           20
```
Comment (alagoutte) [inline PowerArubaCX/Public/vlan.ps1]: same where... (about example)

**Review by alagoutte — CHANGES_REQUESTED** [review]
Need some change before merge

**Comment by JM2K69** [comment]
Closed

**Pull request closed without merging**